src/AutoMapper/Execution/DelegateFactory.cs

```diff
@@ -44,6 +48,15 @@ public static Expression GenerateConstructorExpression(Type type)
                 return Constant(null, typeof(string));
             }
 
+            if (type.IsInterface())
```

Comment:
Is this really necessary?  CollectionMapper returns ICollection it's just casts to List instead of left as instance type.

Reply:
This code is called in a few places so I can't assume what the mappers do.

Reply:
I guess it's not that bad :) What about ISet? Maybe move [arrays](https://github.com/AutoMapper/AutoMapper/blob/605941b79b1f5cf6663c1876660bb9c8bc338cb4/src/AutoMapper/Execution/ExpressionBuilder.cs#L85) here too?

Reply:
I'll get another PR for that for now lgtm